realm/realm-library/src/main/cpp/io_realm_ObjectStoreUserStore.cpp

```diff
@@ -67,11 +79,13 @@ JNIEXPORT void JNICALL Java_io_realm_ObjectStoreUserStore_updateOrCreateUser
  * Method:    logoutCurrentUser
  * Signature: ()V
  */
-JNIEXPORT void JNICALL Java_io_realm_ObjectStoreUserStore_logoutCurrentUser
-        (JNIEnv *, jclass) {
+JNIEXPORT void JNICALL Java_io_realm_ObjectStoreUserStore_logoutCurrentUser (JNIEnv *env, jclass)
+{
     TR_ENTER()
-    const std::shared_ptr<SyncUser>& user = currentUserOrThrow();
-    user->log_out();
+    try {
+        const std::shared_ptr<SyncUser>& user = currentUserOrThrow(env);
```

Comment:
Likewise here. Since `currentUserOrThrow()` just returns, you don't know if `user->log_out()` can be called.